tensorflow_addons/layers/noisy_dense.py

```diff
@@ -0,0 +1,249 @@
+                initializer=self.σ_init,
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+                dtype=self.dtype,
+                trainable=True,
+            )
+
+            self.µ_bias = self.add_weight(
+                "µ_bias",
+                shape=[self.units,],
+                initializer=self.µ_init,
+                regularizer=self.bias_regularizer,
+                constraint=self.bias_constraint,
+                dtype=self.dtype,
+                trainable=True,
+            )
+
+        self.built = True
+
+    @staticmethod
+    def _scale_noise(x):
+        return tf.sign(x) * tf.sqrt(tf.abs(x))
+
+    def call(self, inputs):
+        dtype = self._compute_dtype_object
+        if inputs.dtype.base_dtype != dtype.base_dtype:
+            inputs = tf.cast(inputs, dtype=dtype)
+
+        # Fixed parameters added as the noise
+        ε_i = tf.random.normal([self.last_dim, self.units], dtype=dtype)
+        ε_j = tf.random.normal([self.units,], dtype=dtype)
+
+        # Creates the factorised Gaussian noise
+        f = NoisyDense._scale_noise
+        ε_kernel = f(ε_i) * f(ε_j)
+        ε_bias = f(ε_j)
+
+        # Performs: y = (µw + σw · εw)x + µb + σb · εb
+        # to calculate the output
+        kernel = self.µ_kernel + (self.σ_kernel * ε_kernel)
+
+        if inputs.dtype.base_dtype != dtype.base_dtype:
+            inputs = tf.cast(inputs, dtype=dtype)
+
+        rank = inputs.shape.rank
+        if rank == 2 or rank is None:
+            if isinstance(inputs, tf.sparse.SparseTensor):
+                outputs = tf.sparse.sparse_dense_matmul(inputs, kernel)
+            else:
+                outputs = tf.linalg.matmul(inputs, kernel)
+        # Broadcast kernel to inputs.
+        else:
+            outputs = tf.tensordot(inputs, kernel, [[rank - 1], [0]])
+            # Reshape the output back to the original ndim of the input.
+            if not tf.executing_eagerly():
+                shape = inputs.shape.as_list()
+                output_shape = shape[:-1] + [kernel.shape[-1]]
+                outputs.set_shape(output_shape)
+
+        if self.use_bias:
+            noisy_bias = self.µ_bias + (self.σ_bias * ε_bias)
+            outputs = tf.nn.bias_add(outputs, noisy_bias)
+
+        if self.activation is not None:
+            outputs = self.activation(outputs)
+
+        return outputs
+
+    def compute_output_shape(self, input_shape):
+        input_shape = tf.TensorShape(input_shape)
+        input_shape = input_shape.with_rank_at_least(2)
+        if tf.compat.dimension_value(input_shape[-1]) is None:
+            raise ValueError(
+                "The innermost dimension of input_shape must be defined, but saw: %s"
+                % input_shape
+            )
+        return input_shape[:-1].concatenate(self.units)
+
+    def get_config(self):
+        config = super(NoisyDense, self).get_config()
+        config.update(
```

Comment:
Are you missing something in the config dictionary e.g. like `use_bias` or initializes?

Reply:
>Are you missing something in the config dictionary e.g. like `use_bias`

I added `use_bias` but for some reason, it is now giving me a formatting error even though I have formatted my code multiple times.

>or initializes?

I implemented the initializers used in the paper. If you would like I can make it so that the initializer can be changed to something other than the default, but this may hurt the performance.